api/components/com_media/src/Controller/MediaController.php

```diff
@@ -0,0 +1,401 @@
+
+		// Display files in specific path.
+		if ($this->input->exists('path'))
+		{
+			$this->modelState->set('path', $this->input->get('path', '', 'STRING'));
+		}
+
+		// Return files (not folders) as url's.
+		if ($this->input->exists('url'))
+		{
+			$this->modelState->set('url', $this->input->get('url', true, 'BOOLEAN'));
+		}
+
+		// Map JSON:API compliant filter[search] to com_media model state.
+		$apiFilterInfo = $this->input->get('filter', [], 'array');
+		$filter        = InputFilter::getInstance();
+
+		// Search for files matching (part of) a name or glob pattern.
+		if ($doSearch = array_key_exists('search', $apiFilterInfo))
+		{
+			$this->modelState->set('search', $filter->clean($apiFilterInfo['search'], 'STRING'));
+			// Tell model to search recursively
+			$this->modelState->set('search_recursive', $this->input->get('search_recursive', false, 'BOOLEAN'));
+		}
+
+		return parent::displayList();
+	}
+
+	/**
+	 * Display one specific file or folder.
+	 *
+	 * @param   string  $path  The path of the file to display. Leave empty if you want to retrieve data from the request.
+	 *
+	 * @return  static  A \JControllerLegacy object to support chaining.
+	 *
+	 * @throws  InvalidPathException
+	 * @throws  \Exception
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function displayItem($path = '')
+	{
+		// Set list specific request parameters in model state.
+		$this->setModelState(self::$itemQueryModelStateMap);
+
+		// Display files in specific path.
+		$this->modelState->set('path', $path ?: $this->input->get('path', '', 'STRING'));
+
+		// Return files (not folders) as url's.
+		if ($this->input->exists('url'))
+		{
+			$this->modelState->set('url', $this->input->get('url', true, 'BOOLEAN'));
+		}
+
+		return parent::displayItem();
+	}
+
+	/**
+	 * Set model state using a list of mappings between query parameters and model state names.
+	 *
+	 * @param   array  $mappings  A list of mappings between query parameters and model state names..
+	 *
+	 * @return  void
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	private function setModelState(array $mappings)
+	{
+		foreach ($mappings as $queryName => $modelState)
+		{
+			if ($this->input->exists($queryName))
+			{
+				$this->modelState->set($modelState['name'], $this->input->get($queryName, '', $modelState['type']));
+			}
+		}
+	}
+
+	/**
+	 * Method to add a new file or folder.
+	 *
+	 * @return  void
+	 *
+	 * @throws  FileExistsException
+	 * @throws  InvalidPathException
+	 * @throws  \RuntimeException
+	 * @throws  \Exception
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function add()
+	{
+		$path = $this->input->json->get('path', '', 'STRING');
+		$content = $this->input->json->get('content', '', 'RAW');
+
+		$missingParameters = [];
+
+		if (empty($path))
+		{
+			$missingParameters[] = 'path';
+		}
+
+		if (empty($content))
+		{
+			$missingParameters[] = 'content';
+		}
+
+		if (count($missingParameters))
```

Comment:
Since PHP 7.0, some functions are replaced by opcodes, producing much faster code. 
Yet, for this to work, these functions need to be referenced in the root namespace at compile time:
Here is the exact list of functions that have this behavior:
https://github.com/php/php-src/blob/f2db305fa4e9bd7d04d567822687ec714aedcdb5/Zend/zend_compile.c#L3872

```suggestion
		if (\count($missingParameters))
```